### PR TITLE
huge refactor

### PR DESCRIPTION
**Pull request by nikhan**
This is a huge refactor that migrates the visual graph from SVG to canvas. This provides monumental benefits to performance and scalability. The canvas implementation consists of several buffers that store an image of various elements (nodes, edges) that are all composed together to build the final visual representation of the graph. The canvas implementation can handle thousands of nodes per group -- and probably can be optimized further.

The data structures that govern how the model interacts with the views is now written with a Flux-like architecture in mind.

This PR introduces some regressions (in order of priority):
- group labels cannot be set
- source parameters cannot be set 
- there is no copy and paste
- the group/ungroup buttons are not functional (although you can press `g` and `u`)
- group translations/positions are wonky

### CODE REVIEW

**Comment by nikhan** [comment]
Oh, and we should probably tag the current release...

**Comment by mikedewar** [comment]
made tag

**Comment by mikedewar** [comment]
well, it runs

File: server/group.go [hunk 1]
```diff
@@ -212,10 +214,16 @@ func (s *Server) DeleteGroup(id int) error {
...
+	childrenToDelete := make([]int, len(group.Children), len(group.Children))
+	for i, c := range group.Children {
+		childrenToDelete[i] = c
+	}
+
+	for _, c := range childrenToDelete {
 		if _, ok := s.blocks[c]; ok {
 			err := s.DeleteBlock(c)
 			if err != nil {
+				fmt.Println(err)
```
Comment (mikedewar) [inline server/group.go]: is this for debugging?

**Comment by mikedewar** [comment]
ok, rebase and squash please!

**Comment by mikedewar** [comment]
:green_apple: :green_book: :green_heart:

**Pull request merged**